File: java/InfraredProximitySensor/src/main/java/frc/robot/Robot.java
package frc.robot;

import edu.wpi.first.wpilibj.AnalogInput;
import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

/**
 * This is a sample program demonstrating how to find the value of an Infrared Proximity Sensor
 */

public class Robot extends TimedRobot {

  private final AnalogInput m_infrared = new AnalogInput(0);

  //These min and max values should be tuned for every sensor. Can be changed to refine the distance precision.
  private final double minValue = 1.8;
  private final double maxValue = 2.8;

  @Override
  public void teleopPeriodic() {

    //Get the voltage value from the sensor
    double voltage = m_infrared.getAverageVoltage();
    boolean proximity = false;
    
    //Find if the sensor voltage is within the given range
    if (voltage <= maxValue && voltage >= minValue) {
      proximity = true;
    }

    SmartDashboard.putNumber("Infrared Voltage", voltage);
    SmartDashboard.putBoolean("Proximity Alert", proximity);

  }
}

File: cpp/InfraredProximitySensor/src/main/cpp/Robot.cpp
#include <frc/AnalogInput.h>
#include <frc/MedianFilter.h>
#include <frc/TimedRobot.h>
#include <frc/smartdashboard/SmartDashboard.h>

/**
 * This is a sample program demonstrating how to find the value of an Infrared Proximity Sensor
 */
class Robot : public frc::TimedRobot {
 public:

  void TeleopPeriodic() override {

    //Get the voltage value from the sensor
    double voltage = m_infrared.GetAverageVoltage();
    bool proximity = false;

    //Find if the sensor voltage is within the given range
    if (voltage <= maxValue && voltage >= minValue) {
      proximity = true;
    }

    frc::SmartDashboard::PutNumber("Infrared Voltage", voltage);
    frc::SmartDashboard::PutBoolean("Proximity Alert", proximity);
  }

 private:
  frc::AnalogInput m_infrared{0};

  //These min and max values should be tuned for every sensor. Can be changed to refine the distance precision.
  double minValue = 1.8;
  double maxValue = 2.8;
};

#ifndef RUNNING_FRC_TESTS
int main() { return frc::StartRobot<Robot>(); }
#endif
